.github/workflows/ci-tests.yml

```diff
@@ -23,31 +25,48 @@ jobs:
         with:
           python-version: '3.9'
 
-      - name: Install dependencies
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install missing cached dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction
+
+      - name: Install other dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e ".[tests]"
+          pip install poetry-dynamic-versioning
```

Comment:
I think we should use `poetry plugin add` rather than `pip install` here, as that ensures the plugin is installed into the poetry env and not the project virtual env.

Similarly, I think this command would logically make sense as a separate step immediately after "install poetry". Same for other CI jobs that install poetry. 

More info on that command here:
https://python-poetry.org/docs/master/plugins/

Reply:
This might be possible on future versions of Poetry, but it's not currently supported in the latest 'stable' version.